pylint/checkers/utils.py

```diff
@@ -2281,5 +2281,8 @@ def is_enum_member(node: nodes.AssignName) -> bool:
     ):
         return False
 
-    enum_member_objects = frame.locals.get("__members__")[0].items
+    try:
+        enum_member_objects = frame.locals.get("__members__")[0].items
```

Comment:
Shall we do a check if the `get` returns `None`? In the future I think we will ask ourselves what kind of error we are catching here whereas seeing an `is not None` check is very explicit.

Reply:
I went for try/except since in the majority of cases this code is run, there should be a `__members__` present; the only case I know of where it breaks currently, is when the dataclass decorator is used.
You make a good point though - at first glance it would be a bit difficult to see where the cause of the exception is. We could simplify it:
```diff
+    members = frame.locals.get("__members__")
     try:
-        enum_member_objects = frame.locals.get("__members__")[0].items
+        enum_member_objects = members[0].items
     except TypeError:
         return False
     return node.name in [name_obj.name for value, name_obj in enum_member_objects]
```

Or your suggestion:
```diff
 
-    try:
-        enum_member_objects = frame.locals.get("__members__")[0].items
-    except TypeError:
-        return False
-    return node.name in [name_obj.name for value, name_obj in enum_member_objects]
+    members = frame.locals.get("__members__")
+    if members is not None:
+        enum_member_objects = members[0].items
+        return node.name in [name_obj.name for value, name_obj in enum_member_objects]
+    return False
```

Reply:
I prefer the second diff, but perhaps with an early exit for `if members is None` and then doing the rest of the checks.

We can just add a comment saying that "due to unforeseen circumstances __members__ is not always present" 😄